Sources/AblyAssetTrackingSubscriber/WorkerQueue/SubscriberWorkerQueueProperties.swift

```diff
@@ -3,4 +3,185 @@ import Foundation
+                trackableState = .failed
+            }
+            break
+        case .offline:
+            trackableState = .offline
+        case .failed:
+            trackableState = .failed
+        }
+        
+        if let trackableState = trackableState, trackableState != lastEmittedTrackableState {
+            lastEmittedTrackableState = trackableState
+            notifyTrackableStateUpdated(trackableState: trackableState)
+        }
+        // It is possible for presentPublisherMemberKeys to not be empty, even when there's no connectivity from our side,
+        // because it's possible to have presence entry events without subsequent leave events.
+        // Therefore, from the perspective of a user consuming events from publisherPresenceStateFlow, what matters
+        // is what's computed for isPublisherVisible (not the simple isAPublisherPresent).
+        var isPublisherVisible = false
+        if case .online = lastConnectionStateChange.state {
+            if isAPublisherPresent {
+                isPublisherVisible = true
+            }
+        }
+        
+        if lastEmittedValueOfIsPublisherVisible != isPublisherVisible {
+            lastEmittedValueOfIsPublisherVisible = isPublisherVisible
+            notifyPublisherPresenceUpdated(isPublisherPresent: isPublisherVisible)
+        }
+        
+        notifyResolutionsChanged(resolutions: pendingPublisherResolutions.drain())
+    }
+    
+    mutating func notifyEnhancedLocationUpdated(locationUpdate: LocationUpdate) {
+        enhancedLocation = locationUpdate
+        guard let subscriber = subscriber
+        else { return }
+        delegate?.subscriber(sender: subscriber, didUpdateEnhancedLocation: locationUpdate)
+    }
+    
+    mutating func notifyRawLocationUpdated(locationUpdate: LocationUpdate) {
+        rawLocation = locationUpdate
+        guard let subscriber = subscriber
+        else { return }
+        delegate?.subscriber(sender: subscriber, didUpdateRawLocation: locationUpdate)
+    }
+    
+    mutating func notifyPublisherPresenceUpdated(isPublisherPresent: Bool) {
+        publisherPresence = isPublisherPresent
+        guard let subscriber = subscriber
+        else { return }
+        delegate?.subscriber(sender: subscriber, didUpdatePublisherPresence: publisherPresence)
+    }
+    
+    mutating func notifyTrackableStateUpdated(trackableState: ConnectionState) {
+        self.trackableState = trackableState
+        guard let subscriber = subscriber
+        else { return }
+        delegate?.subscriber(sender: subscriber, didChangeAssetConnectionStatus: trackableState)
+    }
+    
+    mutating func notifyDidFailWithError(error: ErrorInformation) {
+        guard let subscriber = subscriber
+        else { return }
+        delegate?.subscriber(sender: subscriber, didFailWithError: error)
+    }
+    
+    mutating func notifyResolutionsChanged(resolutions: [Resolution]) {
+        if !resolutions.isEmpty {
```

Comment:
Can we use `guard`?

Reply:
9c745748bba921ec848c6eb6225dd9ac5de55579